docs/architecture/api-20201231-preview.md

```diff
@@ -0,0 +1,148 @@
+# API 20201231-preview design
+
+For api `20201231-preview` bellow is the list of proposed changes and features.
+These would have to land into API and backend as a features.
+
+Feature work will need to be agreed and converted to working stories. All technical details bellow are overview only
+
+## API changes
+
+### ComputeSecurityProfile
+
+Add security profile to worker and master pools configuration for all security enhancements
+for the compute instances
+
+We need to make sure that these option are validated on the cluster too when customer is interacting with MachineSet objects. We should be able to verify if these options where enabled on cluster create and set those accordingly. This does not prevent customer to create instances with and without encryption in the same cluster.
+
+```
+// MasterProfile represents a master profile.
+type MasterProfile struct {
+	...
+
+	ComputeSecurityProfile
+}
+
+
+// WorkerProfile represents a worker profile.
+type WorkerProfile struct {
+	...
+
+	ComputeSecurityProfile
+}
+```
+
+// EncryptionAtHostEnum enumerates the values for Encryption at host
+type EncryptionAtHostEnum string
+
+const (
+	// Disabled ...
+	Disabled EncryptionAtHostEnum = "Disabled"
+	// Enabled ...
+	Enabled EncryptionAtHostEnum = "Enabled"
+)
+
+// ComputeSecurityProfile represents an security profile for compute instance
+type ComputeSecurityProfile struct {
+	// EncryptionAtHost defines value encryptionAtHost option for all VirtualMachines.
+	EncryptionAtHost EncryptionAtHostEnum `json:"encryptionAtHost,omitempty"`
+	// / DiskEncryptionSetID defines resourceID for diskEncryptionSet resource. It must be in the same subscription
+	DiskEncryptionSetID string `json:"diskEncryptionSetID,omitempty"`
+}
+
+```
+
+### Extend OpenShiftClusterCredentials to provide kube-config
+
+For backwards compatability we leave `OpenShiftClusterCredentials` as it is.
+In the future we might want to refactor all this to Separate `AccessProfile` to
+be aligned with AKS. But for now we just do simple extension.
+
+Introduce new structure `OpenShiftClusterAdminCredentials` and extend frontend API with one more method to trigger download of the `adminKubeConfig`
+
+```
+// OpenShiftClusterCredentials represents a default an OpenShift cluster's
+// console credentials
+type OpenShiftClusterCredentials struct {
+	...
+}
+
+// OpenShiftClusterAdminCredentials represents an OpenShift cluster's credentials
+type OpenShiftClusterAdminCredentials struct {
```

Comment:
I think I'm +1 on the following design decisions you're making:
* separate struct + API call (can be RBACed differently if we need)
* struct containing []byte